### DISCUSSION TITLE
How to reference itself in embedded

### DISCUSSION THREAD

**Discussion by Break-2020** [mikro-orm/mikro-orm#3145] [Q&A]
Hi,I am a beginner :-)

I have such a requirement in the production environment, how can I write the corresponding code to ensure that there will be no errors?

```
@Embeddable()
export class Component {
  @Embedded(() => Component, { array: true })
  children: Component[] = [];
}
```

If I write like this it produces the following error
`[ExceptionHandler] [error] Maximum call stack size exceeded RangeError: Maximum call stack size exceeded`

**Comment by B4nan** [comment]
What version are you using? How does the full stack trace look like?

**Reply by Break-2020** [reply]
> What version are you using? How does the full stack trace look like?

version is 5.1.3

**Reply by Break-2020** [reply]
> What version are you using? How does the full stack trace look like?

[
![image](https://user-images.githubusercontent.com/73464509/170036725-729e8d53-8214-4c32-85f8-ab36e5919ea6.png)
](url)

**Reply by B4nan** [reply]
Thanks, got it reproduced and it also fails in that method for me. Let me try to fix that quickly, if it's just about this place it should be rather easy.

**Reply by Break-2020** [reply]
Hi,how is the problem going?

**Reply by B4nan** [reply]
It's much more complicated than I initially thought. Will try to review this again next, I am away right now.

**Reply by Break-2020** [reply]
Ok, so is there any other way to solve this kind of scenario that needs to be nested

**Reply by B4nan** [reply]
If you need one (or predefined) level of nesting, create a separate embeddable for each level - extending it should work.

**Reply by Break-2020** [reply]
Maybe I should wait for you to fix it......because its nesting depth is not limited

**Reply by B4nan** [reply]
If you are fine waiting for indefinite time... As said above, this is much harded than anticipated, I don't expect any fix in near future. Definitely not a priority.

Alternative to this is to use JSON properties instead of embeddables. But you will have just POJOs in the data, not class instances (you can use `onLoad` even to create class instances from that. You said its not working for your but that's probably because you kept using the embeddable class instances inside - you can't do that. And you might need to specify the `type: 'json'` explicitly, as otherwise it would act as an array type, not JSON type (and you can store only scalars in array type).

**Reply by Break-2020** [reply]
thank you,I'll try

**Comment by Break-2020** [comment]
[
![image](https://user-images.githubusercontent.com/73464509/170034266-0e290aaa-96d6-4813-b979-1e6f7b25903d.png)
](url)

If I use the second method, it will not report an error, but after saving, children will become ['Object, Object']
